Tests/SwiftParserTest/Assertions.swift

```diff
@@ -29,45 +59,85 @@ import _SwiftSyntaxTestSupport
+        "Expected token kind \(expectedLexeme.tokenKind) but got \(actualLexeme.tokenKind)",
+        file: expectedLexeme.file,
+        line: expectedLexeme.line
+      )
     }
 
-    guard l.leadingTriviaText == r.leadingTriviaText else {
-      return FailStringsEqualWithDiff(
-        String(syntaxText: l.leadingTriviaText),
-        String(syntaxText: r.leadingTriviaText),
-        "Token at index \(idx) does not have matching leading trivia",
-        file: file,
-        line: line
+    if actualLexeme.leadingTriviaText != expectedLexeme.leadingTrivia {
+      FailStringsEqualWithDiff(
+        String(syntaxText: actualLexeme.leadingTriviaText),
+        String(syntaxText: expectedLexeme.leadingTrivia),
+        "Leading trivia does not match",
+        file: expectedLexeme.file,
+        line: expectedLexeme.line
       )
     }
 
-    guard l.tokenText.debugDescription == r.tokenText.debugDescription else {
-      return FailStringsEqualWithDiff(
-        l.tokenText.debugDescription,
-        r.tokenText.debugDescription,
-        "Text at index \(idx) does not have matching text",
-        file: file,
-        line: line
+    if actualLexeme.tokenText.debugDescription != expectedLexeme.tokenText.debugDescription {
+      FailStringsEqualWithDiff(
+        actualLexeme.tokenText.debugDescription,
+        expectedLexeme.tokenText.debugDescription,
+        "Token text does not match",
+        file: expectedLexeme.file,
+        line: expectedLexeme.line
       )
     }
 
-    guard l.trailingTriviaText == r.trailingTriviaText else {
-      return FailStringsEqualWithDiff(
-        String(syntaxText: l.trailingTriviaText),
-        String(syntaxText: r.trailingTriviaText),
-        "Token at index \(idx) does not have matching trailing trivia",
-        file: file,
-        line: line
+    if actualLexeme.trailingTriviaText != expectedLexeme.trailingTrivia {
+      FailStringsEqualWithDiff(
+        String(syntaxText: actualLexeme.trailingTriviaText),
+        String(syntaxText: expectedLexeme.trailingTrivia),
+        "Trailing trivia does not match",
+        file: expectedLexeme.file,
+        line: expectedLexeme.line
       )
     }
+
+    if actualLexeme.flags != expectedLexeme.flags {
+      XCTFail(
+        "Expected flags \(expectedLexeme.flags.debugDescription) but got \(actualLexeme.flags.debugDescription)",
+        file: expectedLexeme.file,
+        line: expectedLexeme.line
+      )
+    }
+  }
+}
+
+func AssertLexemes(
+  _ source: String,
+  lexemes expectedLexemes: [LexemeSpec],
+  file: StaticString = #file,
+  line: UInt = #line
+) {
+  var expectedLexemes = expectedLexemes
+  if expectedLexemes.last?.tokenKind != .eof {
+    expectedLexemes.append(LexemeSpec(.eof, text: ""))
+  }
+  var source = source
+  source.withUTF8 { buf in
+    var lexemes = [Lexer.Lexeme]()
+    for token in Lexer.tokenize(buf, from: 0) {
+      lexemes.append(token)
+
+      guard token.tokenKind != .eof else {
+        break
+      }
```

Comment:
I'd find:
```
if token.tokenKind == .eof {
  break
}
```
much easier to parse, maybe that's just me though 😅

Reply:
I agree but I just copied this from the old `Lexer.lex` method in LexerTests.swift.

I’ll update it in a follow-up PR.